cmd/kube-apiserver/app/server.go

```diff
@@ -241,6 +243,25 @@ func Run(s *options.ServerRunOptions) error {
 		}
 	}
 
+	client, err := internalclientset.NewForConfig(genericConfig.LoopbackClientConfig)
+	if err != nil {
+		kubeAPIVersions := os.Getenv("KUBE_API_VERSIONS")
+		if len(kubeAPIVersions) == 0 {
+			return fmt.Errorf("failed to create clientset: %v", err)
+		}
+
+		// KUBE_API_VERSIONS is used in test-update-storage-objects.sh, disabling a number of API
+		// groups. This leads to a nil client above and undefined behaviour further down.
+		// TODO: get rid of KUBE_API_VERSIONS or define sane behaviour if set
+		glog.Errorf("Failed to create clientset with KUBE_API_VERSIONS=%q. KUBE_API_VERSIONS is only for testing. Things will break.", kubeAPIVersions)
+	}
+
+	// NOTE(ericchang): Internal informers should switch to using
+	// 'pkg/client/informers/informers_generated', the second informer
+	// created here.
+	sharedInformers := informers.NewSharedInformerFactory(nil, client, 10*time.Minute)
+	internalSharedInformers := internalversion.NewSharedInformerFactory(client, 10*time.Minute)
```

Comment:
Yes. I'm tracking my work for that here: https://trello.com/c/kpiFmic9/121-use-generated-shared-informers-listers-everywhere-possible. We also need to decide if the kube-apiserver should switch to using versioned informers instead of internal.